File: src/python/token_types.py
from enum import Enum, auto

class TokenType(Enum):
    NUMBER = auto()
    STRING = auto()
    IDENTIFIER = auto()
    
    KEYWORD_LET = auto()
    KEYWORD_FUNC = auto()
    KEYWORD_IF = auto()
    KEYWORD_ELSE = auto()
    KEYWORD_ELIF = auto()
    KEYWORD_WHILE = auto()
    KEYWORD_FOR = auto()
    KEYWORD_IN = auto()
    KEYWORD_RETURN = auto()
    KEYWORD_BREAK = auto()
    KEYWORD_CONTINUE = auto()
    KEYWORD_TRUE = auto()
    KEYWORD_FALSE = auto()
    KEYWORD_NULL = auto()
    KEYWORD_AND = auto()
    KEYWORD_OR = auto()
    KEYWORD_NOT = auto()
    KEYWORD_CLASS = auto()
    KEYWORD_IMPORT = auto()
    
    PLUS = auto()
    MINUS = auto()
    STAR = auto()
    SLASH = auto()
    PERCENT = auto()
    POWER = auto()
    
    EQUALS = auto()
    EQUALS_EQUALS = auto()
    NOT_EQUALS = auto()
    LESS_THAN = auto()
    GREATER_THAN = auto()
    LESS_EQUALS = auto()
    GREATER_EQUALS = auto()
    
    PLUS_EQUALS = auto()
    MINUS_EQUALS = auto()
    STAR_EQUALS = auto()
    SLASH_EQUALS = auto()
    
    LPAREN = auto()
    RPAREN = auto()
    LBRACE = auto()
    RBRACE = auto()
    LBRACKET = auto()
    RBRACKET = auto()
    
    COMMA = auto()
    DOT = auto()
    COLON = auto()
    SEMICOLON = auto()
    ARROW = auto()
    
    NEWLINE = auto()
    EOF = auto()

KEYWORDS = {
    'let': TokenType.KEYWORD_LET,
    'func': TokenType.KEYWORD_FUNC,
    'if': TokenType.KEYWORD_IF,
    'else': TokenType.KEYWORD_ELSE,
    'elif': TokenType.KEYWORD_ELIF,
    'while': TokenType.KEYWORD_WHILE,
    'for': TokenType.KEYWORD_FOR,
    'in': TokenType.KEYWORD_IN,
    'return': TokenType.KEYWORD_RETURN,
    'break': TokenType.KEYWORD_BREAK,
    'continue': TokenType.KEYWORD_CONTINUE,
    'true': TokenType.KEYWORD_TRUE,
    'false': TokenType.KEYWORD_FALSE,
    'null': TokenType.KEYWORD_NULL,
    'and': TokenType.KEYWORD_AND,
    'or': TokenType.KEYWORD_OR,
    'not': TokenType.KEYWORD_NOT,
    'class': TokenType.KEYWORD_CLASS,
    'import': TokenType.KEYWORD_IMPORT,
}

class Token:
    def __init__(self, token_type, value, line, column):
        self.type = token_type
        self.value = value
        self.line = line
        self.column = column
    
    def __repr__(self):
        return f'Token({self.type}, {self.value!r}, {self.line}:{self.column})'

File: src/rust/src/token_types.rs
use std::collections::HashMap;
use std::fmt;

#[derive(Debug, Clone, PartialEq)]
pub enum TokenType {
    Number,
    String,
    Identifier,
    
    // Keywords
    KeywordLet,
    KeywordFunc,
    KeywordIf,
    KeywordElse,
    KeywordElif,
    KeywordWhile,
    KeywordFor,
    KeywordIn,
    KeywordReturn,
    KeywordBreak,
    KeywordContinue,
    KeywordTrue,
    KeywordFalse,
    KeywordNull,
    KeywordAnd,
    KeywordOr,
    KeywordNot,
    KeywordClass,
    KeywordImport,
    
    // Operators
    Plus,
    Minus,
    Star,
    Slash,
    Percent,
    Power,
    
    // Comparison
    Equals,
    EqualsEquals,
    NotEquals,
    LessThan,
    GreaterThan,
    LessEquals,
    GreaterEquals,
    
    // Compound assignment
    PlusEquals,
    MinusEquals,
    StarEquals,
    SlashEquals,
    
    // Delimiters
    LParen,
    RParen,
    LBrace,
    RBrace,
    LBracket,
    RBracket,
    
    // Punctuation
    Comma,
    Dot,
    Colon,
    Semicolon,
    Arrow,
    
    // Special
    Newline,
    Eof,
}

#[derive(Debug, Clone)]
pub struct Token {
    pub token_type: TokenType,
    pub value: TokenValue,
    pub line: usize,
    pub column: usize,
}

#[derive(Debug, Clone, PartialEq)]
pub enum TokenValue {
    Number(f64),
    String(String),
    Identifier(String),
    Symbol(String),
    None,
}

impl Token {
    pub fn new(token_type: TokenType, value: TokenValue, line: usize, column: usize) -> Self {
        Self {
            token_type,
            value,
            line,
            column,
        }
    }
    
    pub fn number(value: f64, line: usize, column: usize) -> Self {
        Self::new(TokenType::Number, TokenValue::Number(value), line, column)
    }
    
    pub fn string(value: String, line: usize, column: usize) -> Self {
        Self::new(TokenType::String, TokenValue::String(value), line, column)
    }
    
    pub fn identifier(value: String, line: usize, column: usize) -> Self {
        Self::new(TokenType::Identifier, TokenValue::Identifier(value), line, column)
    }
    
    pub fn symbol(token_type: TokenType, value: String, line: usize, column: usize) -> Self {
        Self::new(token_type, TokenValue::Symbol(value), line, column)
    }
    
    pub fn simple(token_type: TokenType, line: usize, column: usize) -> Self {
        Self::new(token_type, TokenValue::None, line, column)
    }
}

impl fmt::Display for Token {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        write!(f, "Token({:?}, {:?}, {}:{})", self.token_type, self.value, self.line, self.column)
    }
}

pub fn create_keywords_map() -> HashMap<String, TokenType> {
    let mut keywords = HashMap::new();
    
    keywords.insert("let".to_string(), TokenType::KeywordLet);
    keywords.insert("func".to_string(), TokenType::KeywordFunc);
    keywords.insert("if".to_string(), TokenType::KeywordIf);
    keywords.insert("else".to_string(), TokenType::KeywordElse);
    keywords.insert("elif".to_string(), TokenType::KeywordElif);
    keywords.insert("while".to_string(), TokenType::KeywordWhile);
    keywords.insert("for".to_string(), TokenType::KeywordFor);
    keywords.insert("in".to_string(), TokenType::KeywordIn);
    keywords.insert("return".to_string(), TokenType::KeywordReturn);
    keywords.insert("break".to_string(), TokenType::KeywordBreak);
    keywords.insert("continue".to_string(), TokenType::KeywordContinue);
    keywords.insert("true".to_string(), TokenType::KeywordTrue);
    keywords.insert("false".to_string(), TokenType::KeywordFalse);
    keywords.insert("null".to_string(), TokenType::KeywordNull);
    keywords.insert("and".to_string(), TokenType::KeywordAnd);
    keywords.insert("or".to_string(), TokenType::KeywordOr);
    keywords.insert("not".to_string(), TokenType::KeywordNot);
    keywords.insert("class".to_string(), TokenType::KeywordClass);
    keywords.insert("import".to_string(), TokenType::KeywordImport);
    
    keywords
}
